### DISCUSSION TITLE
Fade image to background colour

### DISCUSSION THREAD

**Discussion by samparmenter** [libvips/libvips#2769] [Help wanted]
I have an image which I would like to fade to an opacity of 0 to give the impression of a reflection. What is the best way to achieve this. Its essentially going to be a composite of a background and a foreground with the foreground fading out to transparent. Any help would be much appreciated.

**Comment by jcupitt** [comment]
Hi @samparmenter,

There are several of ways of doing this. You can make a gradient image and composite that on top of your image to make a fade, or you can make an alpha that fades to transparency and add that to the image, or you could just scale the pixels to 0.

Are these PNGs with transparent elements? Are you hoping to make a PNG which can be overlaid on different backgrounds? Are you comfortable with the various PDF blending modes (I'd be unsure which to use myself)?

**Reply by samparmenter** [reply]
First of all, thank you for the amazing fast replies. This has to be one of the best supported libraries out there and I can't thank you enough i've had a good look through the discussion and issue areas and you have helped so many people!

I am trying to fade out a transparent PNG so that it can be overlaid on a variety of backgrounds. I have little understanding of PDF blending modes but I am more than happy to play around with them.

**Reply by jcupitt** [reply]
What language are you using? Python's probably the simplest. Make a gradient image and try using composite to layer it over your test image.

**Reply by samparmenter** [reply]
I'm using Elixir but by and large the api is easy enough to reason between Python and Elixir.

**Reply by jcupitt** [reply]
Then multiplying the image alpha by a value in 0-1 might be the simplest. Perhaps:

```python
#!/usr/bin/python3

import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1])

# add an alpha if there is none
if not image.hasalpha():
    image = image.addalpha()

# make a gradient image, 1.0 at the top and 0.0 at the bottom
alpha = pyvips.Image.grey(image.height, image.width).rot270()
rgb = alpha.new_from_image([1, 1, 1])
grad = rgb.bandjoin(alpha).copy(interpretation="srgb")

image = image * grad

image.write_to_file(sys.argv[2])
```

You could add a perspective transform too, but perhaps that's overkill.

**Reply by samparmenter** [reply]
Thats working really nicely. Thank you for all your help!

**Comment by jcupitt** [comment]
To give an example, here's adding a gradient alpha to a JPG image:

```
$ vipsheader nina.jpg 
nina.jpg: 6048x4032 uchar, 3 bands, srgb, jpegload
$ vips grey t1.v 4032 6048 --uchar
$ vips rot t1.v t2.v d270
$ vips bandjoin "nina.jpg t2.v" x.png
```

To make:

![y](https://user-images.githubusercontent.com/580843/164184753-b1964d7e-7ee5-496d-bfc2-f8c8861bef7a.png)
